.github/workflows/optimized-ci-cd.yml

```
@@ -0,0 +1,203 @@
+              repo: context.repo.repo,
+              body: `## 🚀 Preview Deployment Ready!\n\n**URL:** ${{ steps.deploy.outputs.url }}\n\n*Deployed from commit ${{ github.sha }}*`
+            })
+
+  # Health Check
+  health-check:
+    runs-on: ubuntu-latest
+    needs: deploy-vercel
+    if: needs.deploy-vercel.result == 'success'
+    steps:
+      - name: Health check deployment
+        run: |
```

actionlint (1.7.7)

property "url" is not defined in object type {}

(expression)